### PR TITLE
Customizable wallet list

### PR DESCRIPTION
**Pull request by lesliexin**
## ℹ️ What is the context for these changes?

Currently our list of wallets in the modal/bottom sheet supports listing MetaMask, Rainbow, Coinbase, Ledger and generic WalletConnect. We've received feedback requesting that it be possible for other wallets be visibly represented in that list.

With the new `buildConnectors` function, we now support custom connectors to be added of type wagmi `InjectedConnector` or `WalletConnectConnector` and for editing the list of default connectors we currently show.

## 🕹️ Demonstration

**Playground**
Showing different user input cases for a custom wallet
https://user-images.githubusercontent.com/18248358/223806698-a9f2d7bc-2b4a-437e-b266-8105d56f3eec.mov

**CommerceTown**

Desktop
https://user-images.githubusercontent.com/18248358/224116550-b10907b8-2908-4519-b72c-2cad85ac1999.mov

Mobile
https://user-images.githubusercontent.com/18248358/224116249-99847ed1-e4b8-4789-9b65-a292d13d4957.MP4

## 🎩 How can this be tophatted?

- Replace **playground/src/wagmi.ts** with the following code
```
import {buildConnectors} from '@shopify/connect-wallet';
import type {CustomConnector} from '@shopify/connect-wallet';
import {configureChains, createClient} from 'wagmi';
import {mainnet} from 'wagmi/chains';
import {publicProvider} from 'wagmi/providers/public';
import {WalletConnectConnector} from 'wagmi/connectors/walletConnect';

import {Argent} from './assets/Argent';

const {chains, provider, webSocketProvider} = configureChains(
  [mainnet],
  [
    publicProvider(),
  ],
);

const argentConnector: CustomConnector = {
  id: 'argent',
  name: 'Argent',
  connector: new WalletConnectConnector({chains, options: {qrcode: false}}),
  icon: Argent,
  qrCodeSupported: true,
  marketingSite: 'https://www.argent.xyz/',
  mobileAppPrefixes: {
    Android: '',
    iOS: 'https://argent.link/app/wc?uri=',
  },
};

const {connectors, wagmiConnectors} = buildConnectors({
  chains,
  customConnectors: [argentConnector],
});

const client = createClient({
  autoConnect: true,
  connectors: wagmiConnectors,
  provider,
  webSocketProvider,
});

export {chains, client, connectors};
```
- Add [this](https://github.com/Shopify/gm-shop-merchant-app/blob/main/extensions/tokengate-app/src/assets/Argent.tsx) Argent icon asset to an `assets` directory
- Then, you can run locally and try connecting with the QR code to verify the connection works  
Note: If you don't have an Argent wallet, you can still scan and connect with another walletconnect supported wallet (e.g. rainbow)

## ✅ Checklist

- [x] Tested on mobile
- [x] Tested on multiple browsers
- [x] ~~Tested for accessibility~~
- [x] Includes unit tests
- [x] ~~Updated relevant documentation for the changes (if necessary)~~ Will be done in a separate PR in shopify.dev

### CODE REVIEW

**Review by jamiely — APPROVED** [review]
This is all code that has been reviewed before?

**Comment by lesliexin** [comment]
> This is all code that has been reviewed before?

@jamiely Yup, it's all come from already reviewed PRs

File: packages/connect-wallet/src/types/connector.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import {Chain} from '@wagmi/core';
+import {Chain, InjectedConnector} from '@wagmi/core';
```
[the remark this answers is not in the document]
Comment (lesliexin) [reply packages/connect-wallet/src/types/connector.ts:1]: Oh wow nice catch, thank you!

**Pull request merged**